update-dependencies/Program.cs

```diff
@@ -0,0 +1,105 @@
+    {
+        private static readonly string s_repoRoot = Directory.GetCurrentDirectory();
+        private static readonly Config s_config = Config.s_Instance;
+        private static bool s_updateOnly = false;
+
+        public static void Main(string[] args)
+        {
+            if (ParseArgs(args))
+            {
+                DependencyUpdateResults updateResults = UpdateFiles();
+
+                if (!s_updateOnly && updateResults.ChangesDetected())
+                {
+                    CreatePullRequest(updateResults).Wait();
+                }
+            }
+        }
+        
+        private static bool ParseArgs(string[] args)
+        {
+            foreach (string arg in args)
+            {
+                if (string.Equals(arg, "--Update", StringComparison.OrdinalIgnoreCase))
+                {
+                    s_updateOnly = true;
+                }
+                else if (arg.Contains('='))
+                {
+                    int delimiterIndex = arg.IndexOf('=');
+                    string name = arg.Substring(0, delimiterIndex);
+                    string value = arg.Substring(delimiterIndex + 1);
+
+                    Environment.SetEnvironmentVariable(name, value);
+                }
+                else
+                {
+                    Console.Error.WriteLine($"Unrecognized argument '{arg}'");
+                    return false;
+                }
+            }
+
+            return true;
+        }
+
+        private static DependencyUpdateResults UpdateFiles()
+        {
+            IEnumerable<BuildInfo> buildInfos = new[] { BuildInfo.Get("Cli", s_config.CliVersionUrl, fetchLatestReleaseFile: false) };
+            IEnumerable<IDependencyUpdater> updaters = GetUpdaters();
+
+            DependencyUpdater updater = new DependencyUpdater();
+            return updater.Update(updaters, buildInfos);
+        }
+
+        private static Task CreatePullRequest(DependencyUpdateResults updateResults)
+        {
+            string commitMessage = $"Update CLI to {updateResults.UsedBuildInfos.Single().LatestReleaseVersion}";
+
+            GitHubAuth gitHubAuth = new GitHubAuth(s_config.Password, s_config.UserName, s_config.Email);
+
+            PullRequestCreator prCreator = new PullRequestCreator(
+                gitHubAuth,
+                s_config.GitHubProject,
+                s_config.GitHubUpstreamOwner,
+                s_config.GitHubUpstreamBranch,
+                s_config.UserName
+            );
+
+            return prCreator.CreateOrUpdateAsync(commitMessage, commitMessage, string.Empty);
+        }
+
+        private static IEnumerable<IDependencyUpdater> GetUpdaters()
+        {
+            return Directory.GetFiles(s_repoRoot, "Dockerfile", SearchOption.AllDirectories)
+                .Where(path => File.ReadAllText(path).Contains("ENV DOTNET_SDK_VERSION")) 
+                .Select(path => CreateRegexUpdater(path, "Microsoft.DotNet.Cli.Utils"));
```

Comment:
>CreateRegexUpdater(path, "Microsoft.DotNet.Cli.Utils") [](start = 32, length = 54)

I think it might be worthwhile to have a code comment explaining why "Microsoft.DotNet.Cli.Utils" is used versus reading some more abstract information about the CLI.